rclpy/rclpy/context.py

```diff
@@ -187,9 +187,11 @@ def get_domain_id(self) -> int:
             return self.__context.get_domain_id()
 
     def __enter__(self) -> 'Context':
-        # We do not accept parameters here. If one wants to customize the init() call,
-        # they would have to call it manually and not use the ContextManager convenience
-        self.init()
+        if self.__context is None:
+            # This object hasn't been initialized yet, so we can't use a context manager
+            raise RuntimeError(
```

Comment:
What do you think about calling `self.init()` here to keep backwards compatibility?

Reply:
I think I'd be OK with doing that, if we also emit a warning that this usage is deprecated and that `init` should be called before entering the context.

Reply:
And this is now done in [b81046d](https://github.com/ros2/rclpy/pull/1298/commits/b81046d6fe806db44f57385f321f60b59a3babdb)